spec/features/bulk_upload/recommending_trainees_spec.rb

```diff
@@ -30,6 +36,12 @@
     end
 
     context "and I upload a CSV missing dates" do
+      before do
+        allow(BulkUpdate::RecommendationsUploads::ValidateCsvRow).to receive(:new).with(anything).and_return(
```

Comment:
You mentioned this already, but i think we need a better way of providing specific csvs with different errors using factories. Now we have tests for things like "a CSV with one missing date" in bulk-recommend, this set up will fail the rebase.

Reply:
this spec now uses the CSV generator 👍